<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Module 15 - Association mapping: GWAS &amp; sequencing data</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SISG 2023 Module 15</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Sessions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Day 1</li>
    <li>
      <a href="Session01_practical.html">Session 1</a>
    </li>
    <li>
      <a href="Session02_practical.html">Session 2</a>
    </li>
    <li>
      <a href="Session03_practical.html">Session 3</a>
    </li>
    <li>
      <a href="SISGM15_prac4.html">Session 4</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 2</li>
    <li>
      <a href="SISGM15_prac5.html">Session 5</a>
    </li>
    <li>
      <a href="SISGM15_prac6.html">Session 6</a>
    </li>
    <li>
      <a href="Session07_practical.html">Session 7</a>
    </li>
    <li>
      <a href="Session08_practical.html">Session 8</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 3</li>
    <li>
      <a href="SISGM15_prac9.html">Session 9</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/joellembatchou/SISG2023_Association_Mapping">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Module 15 - Association mapping: GWAS &amp;
sequencing data</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-07-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>SISG2023_Association_Mapping/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230530code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230530)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230530code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230530)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2023AssociationMappingtreee832f7a8387538e03b6fa8d5b2cb6c026e057760targetblanke832f7aa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/joellembatchou/SISG2023_Association_Mapping/tree/e832f7a8387538e03b6fa8d5b2cb6c026e057760" target="_blank">e832f7a</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2023AssociationMappingtreee832f7a8387538e03b6fa8d5b2cb6c026e057760targetblanke832f7aa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/joellembatchou/SISG2023_Association_Mapping/tree/e832f7a8387538e03b6fa8d5b2cb6c026e057760" target="_blank">e832f7a</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/sim_rels_geno.bed
    Ignored:    exe/
    Ignored:    mk_website.R
    Ignored:    tmp/

Untracked files:
    Untracked:  .mk_website.R.swp
    Untracked:  analysis/SISGM15_prac4Solution.Rmd
    Untracked:  analysis/SISGM15_prac5Solution.Rmd
    Untracked:  analysis/SISGM15_prac6Solution.Rmd
    Untracked:  analysis/SISGM15_prac9Solution.Rmd
    Untracked:  analysis/Session01_practical_Key.Rmd
    Untracked:  analysis/Session02_practical_Key.Rmd
    Untracked:  analysis/Session03_practical_Key.Rmd
    Untracked:  analysis/Session07_practical_Key.Rmd
    Untracked:  analysis/Session08_practical_Key.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>docs/index.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/joellembatchou/SISG2023_Association_Mapping/blob/f6fc911f9e35da9dc15c810fc81371b3497dcfd8/analysis/index.Rmd" target="_blank">f6fc911</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2023-07-21
</td>
<td>
update lexture files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/joellembatchou/SISG2023_Association_Mapping/c3fd4325c95d362c7dcf45bebcdc399d908f196f/docs/index.html" target="_blank">c3fd432</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2023-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/joellembatchou/SISG2023_Association_Mapping/f9485e62db2104f1691d2af4eb84c39c7004a1c1/docs/index.html" target="_blank">f9485e6</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2023-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/joellembatchou/SISG2023_Association_Mapping/blob/19be6700649fd8fa90d9757757e98b752e216c66/analysis/index.Rmd" target="_blank">19be670</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2023-06-28
</td>
<td>
fix typo
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/joellembatchou/SISG2023_Association_Mapping/blob/15c38cfabcfc72e387994ffbd99d862c5c12821f/analysis/index.Rmd" target="_blank">15c38cf</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2023-06-28
</td>
<td>
add index for website
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Instructors: Joelle Mbatchou &amp; Loic Yengo</p>
<p>You will find on this website link to lecture slides, exercises
including some solutions, and link to the class video recordings.</p>
<p>The official SISG web page can be found <a
href="https://si.biostat.washington.edu/courses/SM2315">here</a>
(requires login) – course evaluations and certificates of completion are
available via this page.</p>
<div id="session-format" class="section level2">
<h2>Session Format</h2>
<p>The module has 10 sessions, each of 90 minutes. The standard format
for a session is approximately:</p>
<ul>
<li>45 minutes of lecture material that will be recorded via Zoom live
and posted at the end of the day</li>
<li>30 minutes of exercises for you to try, with small-group “breakout”
Zoom sessions available, attended by other class participants, and
Teaching Assistants</li>
<li>15 minute discussion of exercises, where the instructors will
present possible solutions and answer questions</li>
</ul>
<p>Please join the module’s <a
href="https://uwbiostatisticssisg.slack.com">Slack channel</a>, where
you can ask questions and see real-time updates from the instructors and
TAs. Each session builds on previous sessions so you will get maximum
benefit by attending every session live and in sequence.</p>
<p>The course schedule is below.</p>
</div>
<div id="pre-requisites" class="section level2">
<h2>Pre-requisites</h2>
<p>We will use an online server during the module to run exercises (see
instructions <a href="using_server.html">here</a> for setup). If you
decide to run analyses from your local machine instead, follow the
instructions below.</p>
<p><strong>Prior to the module</strong>, please install working and
latest versions of:</p>
<ul>
<li>R (Version 4.1+):
<a href="https://www.r-project.org/" target="_blank">https://www.r-project.org/</a></li>
<li>RStudio (Version 1.4+) :
<a href="https://rstudio.com/products/rstudio/download/" target="_blank">https://rstudio.com/products/rstudio/download/</a></li>
<li>PLINK 1.9 :
<a href="https://www.cog-genomics.org/plink2/" target="_blank">https://www.cog-genomics.org/plink2/</a></li>
<li>PLINK 2.0 :
<a href="https://www.cog-genomics.org/plink/2.0/" target="_blank">https://www.cog-genomics.org/plink/2.0/</a></li>
<li>REGENIE :
<a href="https://github.com/rgcgithub/regenie/releases/" target="_blank">https://github.com/rgcgithub/regenie/releases/</a></li>
<li>GCTA :
<a href="https://yanglab.westlake.edu.cn/software/gcta/#Download" target="_blank">https://yanglab.westlake.edu.cn/software/gcta/#Download</a></li>
</ul>
<p>All are free software.</p>
<p>Please be sure to install some necessary R packages with:</p>
<pre class="r"><code>install.packages(&quot;qqman&quot;)
install.packages(&quot;bigsnpr&quot;)
install.packages(&quot;hexbin&quot;)
install.packages(&quot;tidyverse&quot;)
install.packages(&quot;BEDMatrix&quot;)
install.packages(&quot;devtools&quot;)
install.packages(&quot;SKAT&quot;)
install.packages(&quot;patchwork&quot;)
devtools::install_github(&quot;yaowuliu/ACAT&quot;)
if (!require(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
BiocManager::install(&quot;GWASTools&quot;)</code></pre>
</div>
<div id="schedule" class="section level2">
<h2>Schedule</h2>
<p>All times listed below for the schedule are Pacific Daylight Time
(PDT, Seattle time).</p>
<table class="table table-hover table-condensed table-striped" style="margin-left: auto; margin-right: auto;">
<caption>
Monday, July 24th
</caption>
<thead>
<tr>
<th style="text-align:left;">
Time
</th>
<th style="text-align:center;">
Session
</th>
<th style="text-align:left;">
Topic
</th>
<th style="text-align:left;">
Lecture
</th>
<th style="text-align:left;">
Exercises
</th>
<th style="text-align:left;">
Key
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
8:30am-10am
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:left;">
Introduction, Case Control Association Testing &amp; Association Testing
with Quantitative Traits
</td>
<td style="text-align:left;">
<a
href="https://www.dropbox.com/scl/fi/jhphkal6ei3vviqs6mdpd/SISG2023session00.pdf?rlkey=9zwr53aq01zppi5ehvt89bbyo&amp;dl=0">Intro</a>,
<a
href="https://www.dropbox.com/scl/fi/z0ffm3s34dvqfg0h934kn/SISG2023session01.pdf?rlkey=q83u1u6wxc42lbuy5m2ww2ge8&amp;dl=0">Lecture</a>
</td>
<td style="text-align:left;">
<a href="Session01_practical.html">Exercises</a>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
10:30am-12pm
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:left;">
Introduction to the PLINK Software for GWAS &amp; Population Structure
Inference
</td>
<td style="text-align:left;">
<a
href="https://www.dropbox.com/scl/fi/h4yh6mha6mfys59myvw0w/SISG2023session02.pdf?rlkey=p1fecx9td64n84onk2jo1y7t2&amp;dl=0">Lecture</a>
</td>
<td style="text-align:left;">
<a href="Session02_practical.html">Exercises</a>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
1:30pm-3pm
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:left;">
GWAS in Samples with Structure &amp; Introduction to the REGENIE
Software
</td>
<td style="text-align:left;">
<a
href="https://www.dropbox.com/scl/fi/gt5yjftxc6qn6wqvn4x7r/SISG2023session03.pdf?rlkey=fqgwjmreo7bnpbaqgyv6ucbw5&amp;dl=0">Lecture</a>
</td>
<td style="text-align:left;">
<a href="Session03_practical.html">Exercises</a>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
3:30pm-5pm
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:left;">
Power, Sample Size &amp; Design Considerations
</td>
<td style="text-align:left;">
<a
href="https://www.dropbox.com/scl/fi/mwhdnlx95hhsc037j3izc/L4_Power-Design_M15.pdf?rlkey=htf2hyd65e8hhnmcirntz07i4&amp;dl=0">Lecture</a>
</td>
<td style="text-align:left;">
<a href="SISGM15_prac4.html">Exercises</a>
</td>
<td style="text-align:left;">
</td>
</tr>
</tbody>
</table>
<table class="table table-hover table-condensed table-striped" style="margin-left: auto; margin-right: auto;">
<caption>
Tuesday, July 25th
</caption>
<thead>
<tr>
<th style="text-align:left;">
Time
</th>
<th style="text-align:center;">
Session
</th>
<th style="text-align:left;">
Topic
</th>
<th style="text-align:left;">
Lecture
</th>
<th style="text-align:left;">
Exercises
</th>
<th style="text-align:left;">
Key
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
8:30am-10am
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:left;">
Downstream analyses: Conditional Analyses, Colocalization, Fine-mapping,
Gene-based &amp; Pathways Analysis
</td>
<td style="text-align:left;">
<a
href="https://www.dropbox.com/scl/fi/e3idft2ibdhewpeo887ru/L5_Downstream-Analyses_M15.pdf?rlkey=b9295tc25icy0xzcjtx3abaag&amp;dl=0">Lecture</a>
</td>
<td style="text-align:left;">
<a href="SISGM15_prac5.html">Exercises</a>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
10:30am-12pm
</td>
<td style="text-align:center;">
6
</td>
<td style="text-align:left;">
Downstream analyses: Polygenic Risk Scores
</td>
<td style="text-align:left;">
<a
href="https://www.dropbox.com/scl/fi/1uh8ee1rj02vtuqqyh24i/L6_Polygenic_Scores_Lectures_M15.pdf?rlkey=pcej9onlpg2rwn7g1kzx5ukt6&amp;dl=0">Lecture</a>
</td>
<td style="text-align:left;">
<a href="SISGM15_prac6.html">Exercises</a>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
1:30pm-3pm
</td>
<td style="text-align:center;">
7
</td>
<td style="text-align:left;">
Rare Variant Analysis: Collapsing Tests, Kernel (Variance Component)
Tests and Omnibus Tests
</td>
<td style="text-align:left;">
<a
href="https://www.dropbox.com/scl/fi/b8d1jxaz1lti0jc2ykcyi/SISG2023session07.pdf?rlkey=bz8x7cffh5gkdtfnr88y2yl18&amp;dl=0">Lecture</a>
</td>
<td style="text-align:left;">
<a href="Session07_practical.html">Exercises</a>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
3:30pm-5pm
</td>
<td style="text-align:center;">
8
</td>
<td style="text-align:left;">
Emerging issues showcasing ongoing research
</td>
<td style="text-align:left;">
<a
href="https://www.dropbox.com/scl/fi/rx1ibf1gp7f7i0jgf5as8/SISG2023session08.pdf?rlkey=xbl1oblyndjtist0cytm500hc&amp;dl=0">Lecture</a>
</td>
<td style="text-align:left;">
<a href="Session08_practical.html">Exercises</a>
</td>
<td style="text-align:left;">
</td>
</tr>
</tbody>
</table>
<table class="table table-hover table-condensed table-striped" style="margin-left: auto; margin-right: auto;">
<caption>
Wednesday, July 26th
</caption>
<thead>
<tr>
<th style="text-align:left;">
Time
</th>
<th style="text-align:center;">
Session
</th>
<th style="text-align:left;">
Topic
</th>
<th style="text-align:left;">
Lecture
</th>
<th style="text-align:left;">
Exercises
</th>
<th style="text-align:left;">
Key
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
8:30am-10am
</td>
<td style="text-align:center;">
9
</td>
<td style="text-align:left;">
Beyond traditional GWAS: Variance QTL, Interaction Testing, GWAX,
Time-to-event, Multi-trait analysis
</td>
<td style="text-align:left;">
<a
href="https://www.dropbox.com/scl/fi/pra6xf0r7vwgsb435povy/L9_Beyond_Standard_GWAS_M15.pdf?rlkey=ewnfj5kl9jpcjsvxpyfsf3b9y&amp;dl=0">Lecture</a>
</td>
<td style="text-align:left;">
<a href="SISGM15_prac9.html">Exercises</a>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
10:30am-12pm
</td>
<td style="text-align:center;">
10
</td>
<td style="text-align:left;">
Emerging issues showcasing ongoing research
</td>
<td style="text-align:left;">
Lecture
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
</tbody>
</table>
</div>
<div id="datasets" class="section level2">
<h2>Datasets</h2>
<p>All the data sets used can be downloaded from the <a
href="https://github.com/joellembatchou/SISG2023_Association_Mapping/tree/master/data">Github
repository</a>. These have all been loaded on the online server for the
class.</p>
</div>
<div id="additional-resources" class="section level2">
<h2>Additional resources</h2>
<ul>
<li><a
href="https://faculty.washington.edu/tathornt/sisg/Zheng_Weir_Eigenanalysis_TPB_2016.pdf">2016
TPB Paper</a> on PCA for Population Structure Inference and Admixture
Estimation</li>
<li><a
href="https://faculty.washington.edu/tathornt/sisg/HapMapMXL_Ancestry_Supervised.pdf">HapMap
MXL Individual Ancestry Figure</a> (Supervised Learning)</li>
<li>If you are new to R, you may find the following material helpful:
<ul>
<li><a
href="https://faculty.washington.edu/kenrice/rintro/">Introduction to R
materials</a> from SISG Module 3</li>
<li><a href="https://ggplot2.tidyverse.org/">Graphics with
ggplot2</a></li>
<li><a href="http://dplyr.tidyverse.org/">Data manipulation with
dplyr</a></li>
</ul></li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_1.1.2

loaded via a namespace (and not attached):
 [1] jsonlite_1.8.5    compiler_4.3.0    promises_1.2.0.1  webshot_0.5.5    
 [5] tidyselect_1.2.0  Rcpp_1.0.10       xml2_1.3.4        stringr_1.5.0    
 [9] git2r_0.32.0      later_1.3.1       jquerylib_0.1.4   systemfonts_1.0.4
[13] scales_1.2.1      yaml_2.3.7        fastmap_1.1.1     R6_2.5.1         
[17] generics_0.1.3    workflowr_1.7.0   knitr_1.43        tibble_3.2.1     
[21] kableExtra_1.3.4  munsell_0.5.0     rprojroot_2.0.3   svglite_2.1.1    
[25] bslib_0.5.0       pillar_1.9.0      rlang_1.1.1       utf8_1.2.3       
[29] cachem_1.0.8      stringi_1.7.12    httpuv_1.6.11     xfun_0.39        
[33] fs_1.6.2          sass_0.4.6        viridisLite_0.4.2 cli_3.6.1        
[37] magrittr_2.0.3    rvest_1.0.3       digest_0.6.31     rstudioapi_0.14  
[41] lifecycle_1.0.3   vctrs_0.6.2       evaluate_0.21     glue_1.6.2       
[45] whisker_0.4.1     colorspace_2.1-0  fansi_1.0.4       httr_1.4.6       
[49] rmarkdown_2.22    tools_4.3.0       pkgconfig_2.0.3   htmltools_0.5.5  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
