<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="" />


<title>Session 08 - Exercises</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SISG 2023 Module 15</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Sessions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Day 1</li>
    <li>
      <a href="Session01_practical.html">Session 1</a>
    </li>
    <li>
      <a href="Session02_practical.html">Session 2</a>
    </li>
    <li>
      <a href="Session03_practical.html">Session 3</a>
    </li>
    <li>
      <a href="SISGM15_prac4.html">Session 4</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 2</li>
    <li>
      <a href="SISGM15_prac5.html">Session 5</a>
    </li>
    <li>
      <a href="SISGM15_prac6.html">Session 6</a>
    </li>
    <li>
      <a href="Session07_practical.html">Session 7</a>
    </li>
    <li>
      <a href="Session08_practical.html">Session 8</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 3</li>
    <li>
      <a href="SISGM15_prac9.html">Session 9</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/joellembatchou/SISG2023_Association_Mapping">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Session 08 - Exercises</h1>
<h4 class="author"></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-07-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>SISG2023_Association_Mapping/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230530code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230530)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230530code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230530)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2023AssociationMappingtree0e55c6d748ae30da0afa81c7d7564c44ff47d56ftargetblank0e55c6da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/joellembatchou/SISG2023_Association_Mapping/tree/0e55c6d748ae30da0afa81c7d7564c44ff47d56f" target="_blank">0e55c6d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2023AssociationMappingtree0e55c6d748ae30da0afa81c7d7564c44ff47d56ftargetblank0e55c6da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/joellembatchou/SISG2023_Association_Mapping/tree/0e55c6d748ae30da0afa81c7d7564c44ff47d56f" target="_blank">0e55c6d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/SISGM15_prac5Solution.Rmd
    Untracked:  analysis/SISGM15_prac6Solution.Rmd
    Untracked:  analysis/SISGM15_prac9Solution.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/Session08_practical_Key.Rmd</code>) and HTML
(<code>docs/Session08_practical_Key.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/joellembatchou/SISG2023_Association_Mapping/blob/0e55c6d748ae30da0afa81c7d7564c44ff47d56f/analysis/Session08_practical_Key.Rmd" target="_blank">0e55c6d</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2023-07-24
</td>
<td>
update s8 key
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Before you begin:</p>
<ul>
<li>Make sure that R is installed on your computer</li>
<li>For this lab, we will use the following R libraries:</li>
</ul>
<pre class="r"><code>library(data.table)
library(dplyr)
library(tidyr)
library(GWASTools)
library(ggplot2)
library(patchwork)</code></pre>
<p>The R template to do the exercises is <a
href="https://github.com/joellembatchou/SISG2023_Association_Mapping/tree/master/code">here</a>.</p>
<div id="case-control-imbalance-in-gwas" class="section level2">
<h2>Case-control imbalance in GWAS</h2>
<div id="introduction" class="section level3">
<h3>Introduction</h3>
<p>We will use a simulated dataset consisting of 3 binary traits with
different amounts of case-control imbalance, as well as a genetic data
set of null SNPs to examine the null distribution of the test statistics
from GWAS.</p>
</div>
<div id="simulate-the-data" class="section level3">
<h3>Simulate the data</h3>
<p>We can use PLINK1.9 to simulate the genetic dataset. For <span
class="math inline">\(N=10,000\)</span> samples, Let’s simulate 10,000
variants where 5,000 are common with MAF chosen from a Uniform(0.05,
0.5) distribution and for the rare variants, we will use a
Uniform(0.001, 0.01) distribution. Run the following command in R:</p>
<pre class="r"><code>N &lt;- 10e3
# Generate a configuration file specifying allele frequencies (a,b) for Uniform(a,b) distribution
write(paste0(&quot;5000 common 0.05 0.5 1 1&quot;), &quot;sim.config&quot;)
write(paste0(&quot;5000 rare 0.001 0.01 1 1&quot;), &quot;sim.config&quot;, append = TRUE)
# Run PLINK1.9
system(paste0(&quot;/data/SISG2023M15/exe/plink --make-bed --simulate sim.config --simulate-ncases &quot;, N, &quot; --simulate-ncontrols 0 --simulate-prevalence 0.1  --out cc_imb_geno&quot;))</code></pre>
<p>You should now have files <code>cc_imb_geno.{bed,bim,fam}</code>.</p>
<p>For the phenotype data simulation, we will simulate 3 phenotypes with
different levels of case-control imbalance (CCR 1:9, 1:99, and 1:199).
Run the following code</p>
<pre class="r"><code># get FID/IID from FAM file
sample.ids &lt;- fread(&quot;cc_imb_geno.fam&quot;, header = FALSE)
N &lt;- nrow(sample.ids)

## Set prevalence = 10% (CCR 1:9)
y1 &lt;- rbinom(N, 1, prob = 0.1 )
## Set prevalence = 1% (CCR 1:99)
y2 &lt;- rbinom(N, 1, prob = 0.01 )
## Set prevalence = 0.5% (CCR 1:199)
y3 &lt;- rbinom(N, 1, prob = 0.005 )

# write to file
data.frame(FID = sample.ids$V1, IID = sample.ids$V2, Y1 = y1, Y2 = y2, Y3 = y3) %&gt;%
  fwrite(&quot;cc_imb_pheno.txt&quot;, sep = &quot;\t&quot;, na = NA, quote = FALSE)</code></pre>
<p>You should now have file <code>cc_imb_pheno.txt</code>.</p>
</div>
<div id="exercises" class="section level3">
<h3>Exercises</h3>
<p>We will assess the null distribution of our test statistics when
performing association mapping using different models. Here are some
things to try:</p>
<ol style="list-style-type: decimal">
<li>Run GWAS in REGENIE (step 2) analyzing all 3 traits as quantitative.
The basic command would be</li>
</ol>
<pre class="r"><code>system(&quot;/data/SISG2023M15/exe/regenie --bed cc_imb_geno --phenoFile cc_imb_pheno.txt --step 2 --bsize 400 --qt  --force-qt --ignore-pred --out regenie_step2&quot;)</code></pre>
<p>This will produce three files (one for each phenotype):
<code>&lt;output_prefix&gt;_Y1.regenie</code>,
<code>&lt;output_prefix&gt;_Y2.regenie</code>,
<code>&lt;output_prefix&gt;_Y3.regenie</code></p>
<ol start="2" style="list-style-type: decimal">
<li>Read in the three summary statistics files in R and make a QQ plot
of the p-values for each phenotype. Since these are null SNPs, how does
it compare to what we expect? The basic command for one phenotype would
be</li>
</ol>
<pre class="r"><code>sumstats.y1 &lt;- fread(&quot;regenie_step2_Y1.regenie&quot;) %&gt;% mutate(Pval = 10^(-LOG10P), Z = sign(BETA) * sqrt(CHISQ))
qqPlot( pval = sumstats.y1$Pval )</code></pre>
<p><img src="figure/Session08_practical_Key.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sumstats.y2 &lt;- fread(&quot;regenie_step2_Y2.regenie&quot;) %&gt;% mutate(Pval = 10^(-LOG10P), Z = sign(BETA) * sqrt(CHISQ))
qqPlot( pval = sumstats.y2$Pval )</code></pre>
<p><img src="figure/Session08_practical_Key.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sumstats.y3 &lt;- fread(&quot;regenie_step2_Y3.regenie&quot;) %&gt;% mutate(Pval = 10^(-LOG10P), Z = sign(BETA) * sqrt(CHISQ))
qqPlot( pval = sumstats.y3$Pval )</code></pre>
<p><img src="figure/Session08_practical_Key.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<ol start="3" style="list-style-type: decimal">
<li>Make a histogram of the test statistics for each phenotype and
overlay with a normal distribution. How well do they match?</li>
</ol>
<pre class="r"><code>plot.sumstats.hist &lt;- function(df, title = &quot;&quot;){
  df %&gt;%
  ggplot( aes(x = Z) ) +
  geom_histogram(aes(y = ..density..), colour=&quot;black&quot;, fill=&quot;white&quot;, bins = 100) +
  stat_function(
    fun = dnorm, 
    col = &quot;red&quot;,
    args = list(mean = mean(df$Z), sd = sd(df$Z))
  ) +
    labs(title = title)
}

# for Y1
plot.sumstats.hist(sumstats.y1, title = &quot;Y1&quot;)</code></pre>
<pre><code>Warning: The dot-dot notation (`..density..`) was deprecated in ggplot2 3.4.0.
ℹ Please use `after_stat(density)` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<p><img src="figure/Session08_practical_Key.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># for Y2
plot.sumstats.hist(sumstats.y2, title = &quot;Y2&quot;)</code></pre>
<p><img src="figure/Session08_practical_Key.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># For Y3
plot.sumstats.hist(sumstats.y3, title = &quot;Y3&quot;)</code></pre>
<p><img src="figure/Session08_practical_Key.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>What do you observe as the case-control imbalance gets more
severe?</p>
<ol start="4" style="list-style-type: decimal">
<li>Re-do 3 but now separate the histogram for common and rare
SNPs.</li>
</ol>
<ul>
<li>First separate the data frame based on common/rare simulated
SNPs.</li>
</ul>
<pre class="r"><code>sumstats.y1.common &lt;- sumstats.y1[ grepl(&quot;common&quot;, ID), ]
sumstats.y1.rare &lt;- sumstats.y1[ grepl(&quot;rare&quot;, ID), ]
sumstats.y2.common &lt;- sumstats.y2[ grepl(&quot;common&quot;, ID), ]
sumstats.y2.rare &lt;- sumstats.y2[ grepl(&quot;rare&quot;, ID), ]
sumstats.y3.common &lt;- sumstats.y3[ grepl(&quot;common&quot;, ID), ]
sumstats.y3.rare &lt;- sumstats.y3[ grepl(&quot;rare&quot;, ID), ]</code></pre>
<ul>
<li>Make a histogram of the test statistics distribution at common/rare
SNPs. What do you observe across the different case-control
imbalances?</li>
</ul>
<pre class="r"><code>plot.sumstats.hist(sumstats.y1.common, title = &quot;Y1 - Common SNPs&quot;) | plot.sumstats.hist(sumstats.y1.rare, title = &quot;Y1 - Rare SNPs&quot;) </code></pre>
<p><img src="figure/Session08_practical_Key.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.sumstats.hist(sumstats.y2.common, title = &quot;Y2 - Common SNPs&quot;) | plot.sumstats.hist(sumstats.y2.rare, title = &quot;Y2 - Rare SNPs&quot;) </code></pre>
<p><img src="figure/Session08_practical_Key.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.sumstats.hist(sumstats.y3.common, title = &quot;Y3 - Common SNPs&quot;) | plot.sumstats.hist(sumstats.y3.rare, title = &quot;Y3 - Rare SNPs&quot;) </code></pre>
<p><img src="figure/Session08_practical_Key.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.2 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] patchwork_1.1.2     ggplot2_3.4.2       GWASTools_1.46.0   
[4] Biobase_2.60.0      BiocGenerics_0.46.0 tidyr_1.3.0        
[7] dplyr_1.1.2         data.table_1.14.8  

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0     farver_2.1.1         blob_1.2.4          
 [4] gdsfmt_1.36.1        fastmap_1.1.1        promises_1.2.0.1    
 [7] digest_0.6.33        rpart_4.1.19         lifecycle_1.0.3     
[10] survival_3.5-5       RSQLite_2.3.1        magrittr_2.0.3      
[13] compiler_4.3.1       rlang_1.1.1          sass_0.4.6          
[16] tools_4.3.1          utf8_1.2.3           yaml_2.3.7          
[19] knitr_1.43           labeling_0.4.2       bit_4.0.5           
[22] withr_2.5.0          workflowr_1.7.0      purrr_1.0.1         
[25] GWASExactHW_1.01     nnet_7.3-19          grid_4.3.1          
[28] fansi_1.0.4          git2r_0.32.0         jomo_2.7-6          
[31] colorspace_2.1-0     mice_3.16.0          scales_1.2.1        
[34] iterators_1.0.14     MASS_7.3-60          cli_3.6.1           
[37] crayon_1.5.2         rmarkdown_2.23       generics_0.1.3      
[40] rstudioapi_0.15.0    minqa_1.2.5          DBI_1.1.3           
[43] DNAcopy_1.74.1       cachem_1.0.8         stringr_1.5.0       
[46] operator.tools_1.6.3 splines_4.3.1        vctrs_0.6.3         
[49] boot_1.3-28          glmnet_4.1-7         Matrix_1.5-4.1      
[52] sandwich_3.0-2       SparseM_1.81         jsonlite_1.8.7      
[55] bit64_4.0.5          quantsmooth_1.66.0   mitml_0.4-5         
[58] foreach_1.5.2        jquerylib_0.1.4      glue_1.6.2          
[61] nloptr_2.0.3         pan_1.8              codetools_0.2-19    
[64] gtable_0.3.3         stringi_1.7.12       shape_1.4.6         
[67] later_1.3.1          munsell_0.5.0        lmtest_0.9-40       
[70] lme4_1.1-34          tibble_3.2.1         pillar_1.9.0        
[73] htmltools_0.5.5      quantreg_5.95        R6_2.5.1            
[76] formula.tools_1.7.1  rprojroot_2.0.3      evaluate_0.21       
[79] lattice_0.21-8       highr_0.10           backports_1.4.1     
[82] memoise_2.0.1        broom_1.0.5          httpuv_1.6.11       
[85] bslib_0.5.0          MatrixModels_0.5-1   Rcpp_1.0.11         
[88] nlme_3.1-162         mgcv_1.8-42          logistf_1.25.0      
[91] whisker_0.4.1        xfun_0.39            fs_1.6.2            
[94] zoo_1.8-12           pkgconfig_2.0.3     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
